File: Pascal/bbb.pas
// bbb: Check output from the BEDBUG512 CSPRNG 
program bbb;
uses bb512;
// mixing rounds: 7 is recommended minimum
const ROUNDS = 7;
var n: Cardinal;
	seed: string = 'Bacon';
begin
	// pull a seed from the command line
	if ParamCount>0 then seed:=ParamStr(1);
	// seed BEDBUG and mix
	bbSeedW(seed,STSZ*ROUNDS);
	// show some MOD 26 stream output
	for n:=1 to 70 do
		write(chr(bbRandom mod 26 + ord('A')));
	writeln;
end.

File: C/bbb.c
// bbb: Check output from the BEDBUG512 CSPRNG 
#include "bb512.h"
#include <stdio.h>
#include <stddef.h>
// mixing rounds: 7 recommended minimum
#define ROUNDS 7

int main(int argc, char *argv[]) {
	char seed[2048]="Bacon";
	// pull a seed from the command line
	if (argc>=2) strcpy(seed,argv[1]);
	// seed BEDBUG and mix
	bb512_SeedW(seed,512*ROUNDS);
	int i;
	// show some MOD 26 stream output
	for (i=0; i<70; i++) printf("%c",bb512_Random() % 26 + 'A');
	printf("\n");
	return 0;
}
